### PR TITLE
Implemented namedtuple, lint indicator and some minor cleanup.

### PR DESCRIPTION
**Pull request by pykong**
This PR addresses: 
#709  
#711 
#736

### CODE REVIEW

**Comment by braver** [comment]
@pykong can you pull `next` into your branch and resolve the merge conflicts?

**Comment by pykong** [comment]
Is mergeable now.

File: lint/linter.py [hunk 1]
```diff
@@ -15,6 +16,18 @@
...
+    "match": None,
+    "line": None,
+    "col": None,
+    "error": None,
+    "warning": None,
+    "message": '',
+    "near": None
+})
+
+LintMatch = namedtuple("LintMatch", MATCH_DICT.keys())
```
Comment (FichteFoll) [inline lint/linter.py]: You can provide default arguments by assigning a tuple to `LintMatch.__new__.__defaults__`, i.e. 

```py
LintMatch = namedtuple("LintMatch", MATCH_DICT.keys())
LintMatch.__new__.__defaults__ = tuple(MATCH_DICT.values())
```

File: lint/linter.py [hunk 2]
```diff
@@ -1628,27 +1651,28 @@ def split_match(self, match):
...
                     col = int(col) - self.line_col_base[1]
                 else:
                     col = len(col)
+                match_dict["col"] = col
 
-            return match, line, col, error, warning, message, near
-        else:
-            persist.debug('No match for regex: {}'.format(self.regex.pattern))
-            return match, None, None, None, None, '', None
+        return LintMatch(*match_dict.values())
```
Comment (FichteFoll) [inline lint/linter.py]: Could also use `LintMatch(**match_dict)` here. It's an ordered dict anyway, but this seems more explicit to me (despite having fewer characters).

Comment (pykong) [reply lint/linter.py]: Thanks. Didn't know that one!

File: lint/linter.py (hunk 1 above)
Comment (pykong) [reply lint/linter.py]: Thanks. Didn't know that one!

File: lint/linter.py (hunk 2 above)
Comment (pykong) [reply lint/linter.py]: Agreed.

**Comment by pykong** [comment]
Closing this PR until branch is merge ready again, as in the meantime I have added further changes.

**Comment by pykong** [comment]
Reopened.

**Comment by pykong** [comment]
Closed. Again, as we decided to move jsonschema into a sperate package.

**Comment by pykong** [comment]
Reopening after removing jsonschema.

**Comment by braver** [comment]
Lol, yeah, this looks kinda messy. I prefer it this way though, because it gives me a single iteration to test and we don't have to deal with conflicting or overlapping changes. Once you know the rules well enough you get to break them ;).

File: lint/linter.py [hunk 3]
```diff
@@ -1050,9 +1064,9 @@ def get_cmd(self):
         # For backwards compatibility: SL3 allowed a '@python' suffix which,
         # when set, triggered special handling. SL4 doesn't need this marker,
         # bc all the special handling is just done in the subclass.
-        which = cmd[0]
-        if '@python' in which:
-            cmd[0] = which[:which.find('@python')]
+        # which = cmd[0]
+        # if '@python' in which:
+        #     cmd[0] = which[:which.find('@python')]
```
Comment (braver) [inline lint/linter.py]: @pykong should this be commented out?

File: sublime_linter.py [hunk 4]
```diff
@@ -316,6 +300,10 @@ def highlight(self, view, linters, hit_time):
             panel.fill_panel(window, update=True)
 
         for view in window_views.values():
+            # stop lint indicator
+            ind = persist.indicators[view.id()]
+            ind.stop()
```
Comment (braver) [inline sublime_linter.py]: I dislike short variable names if the abbreviation isn't common (e.g. "img" is fine). This particular one (and a similar one a few lines down) should also work without introducing a variable right?

Comment (pykong) [reply sublime_linter.py]: I think those are fine as long as the abbreviation can be deduced from context.
In this case it is clear `ind` stands for *ind*icator. Nonetheless have changed it to a longer variable name. Yes, we could just chain the `start()`/`stop()` method, but I find it less readable.

**Comment by braver** [comment]
The W/E counters in the statusbar now blink on every change: the numbers disappear for a split second, and then re-appear with the new value.

**Comment by pykong** [comment]
I see what I can do about that blinking.

**Comment by braver** [comment]
Also, sometimes no lint errors show up, not even after running "lint all views". I'm running in debug and see lot's of output but the counters stay at zero and the panel is empty as well...

**Comment by kaste** [comment]
I don't think the indicator should be part of this PR.

Note: If you patch `hit`, the indicator starts if we record (queue) the hit. The queue callback `lint` seems better. But `Linter.lint_view` can still return early so that its callback `highlight` is never called.

File: lint/linter.py (hunk 1 above)
Comment (pykong) [reply lint/linter.py]: @FichteFoll  `.values()` ? Shouldn't it be keys?

File: lint/linter.py [hunk 5]
```diff
@@ -15,6 +16,18 @@
 ARG_RE = re.compile(r'(?P<prefix>@|--?)?(?P<name>[@\w][\w\-]*)(?:(?P<joiner>[=:])(?:(?P<sep>.)(?P<multiple>\+)?)?)?')
 BASE_CLASSES = ('PythonLinter',)
 
+MATCH_DICT = OrderedDict({
```
Comment (pykong) [inline lint/linter.py]: Who spots the problem here? :-)

**Comment by pykong** [comment]
Closing this PR, as branch `next_namedtuple` has been merged into branch `braver`.

**Pull request closed without merging**